abce/__init__.py

```diff
@@ -717,8 +591,12 @@ def build_agents(self, AgentClass, group_name, number=None, parameters={}, agent
 
             self.family_list[group_name].append(family)
             self.num_of_agents_in_group[group_name] = num_agents_this_group
+        return Group(self, self.family_list[group_name], group_name)
 
-    def add_agents(self, messages, round):
+    def add_agents(self, round):
+        messages = self._agents_to_add
+        if len(messages) == 0:
+            return
```

Comment:
yes but why do you test that len(mhessages) is 0? If the list is empty the the following for loop would anyway not do anything?